File: samples/spring-mvc-boot-3-validation-java/src/main/java/io/openapiprocessor/samples/FooController.java
package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.FooApi;
import io.openapiprocessor.openapi.model.Foo;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RestController;

@Validated
@RestController
public class FooController implements FooApi {

    @Override
    public Foo postFoo (Foo foo) {
        return foo;
    }
}

File: samples/spring-mvc-boot-3-monolith-kt/src/main/kotlin/io/openapiprocessor/samples/foo/api/FooController.kt
package io.openapiprocessor.samples.foo.api

import io.openapiprocessor.samples.foo.domain.FooService
import io.openapiprocessor.samples.foo.model.FooDto
import org.springframework.web.bind.annotation.RestController

@RestController
class FooController(val mapper: FooMapper, val service: FooService): FooApi {

    override fun postFoo(foo: FooDto): FooDto {
        return mapper.map(service.echo(mapper.map(foo)))
    }
}
